scapy/layers/inet.py

```diff
@@ -1003,8 +1003,8 @@ def _packetlist_timeskew_graph(self, ip, **kargs):
     """Tries to graph the timeskew between the timestamps and real time for a given ip"""
 
     # Filter TCP segments which source address is 'ip'
-    res = [self._elt2pkt(x) for x in self.res]
-    b = filter(lambda x:x.haslayer(IP) and x.getlayer(IP).src == ip and x.haslayer(TCP), res)
+    res = (self._elt2pkt(x) for x in self.res)
```

Comment:
Maybe you could replace those two lines with:
```Python
b = (x for x in (self._elt2pkt(x) for x in self.res) if IP in x and x[IP].src == ip and TCP in x)
```
Or, if you want to keep two lines for readability, at least replace the name of the temporary variable:
```Python
tmp = (self._elt2pkt(x) for x in self.res)
b = (x for x in tmp if IP in x and x[IP].src == ip and TCP in x)
```
In both cases, I think it's better to use the "new" syntax (`IP in x` and `x[IP]`) rather than the `.{has,get}layer()` methods.